src/stores/database/postgres/ethereum.go

```diff
@@ -99,6 +100,35 @@ func (ea *ETHAccounts) GetAllDeleted(ctx context.Context) ([]*entities.ETHAccoun
 	return accounts, nil
 }
 
+func (s *ETHAccounts) ListAddresses(ctx context.Context, isDeleted bool, limit, offset int) ([]string, error) {
+	var ids = []string{}
+	var err error
+	var query string
+	args := []interface{}{s.storeID}
+
+	switch {
+	case limit != 0 || offset != 0:
```

Comment:
can we have a generic function in `infra` to do this ? Like one accepting the id field name (address or id), limit and offset and returns a list of string? The function looks very similar in all 3 agents

Reply:
I also considered but the complexity of the method it would be too high for not much value as we alredy know this won't be extended further than this 3 agents. Although happy to refactor if you think otherwise on this

Reply:
It could be extended to aliases, roles, nodes and stores. You can list those by id as well even if it's unlikely at the moment that we implement this.

Reply:
Okay, done